src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -819,7 +819,7 @@ impl<'a, 'tcx> DocFolder for LinkCollector<'a, 'tcx> {
                             // The resolved item did not match the disambiguator; give a better error than 'not found'
                             let msg = format!("unresolved link to `{}`", path_str);
                             report_diagnostic(cx, &msg, &item, &dox, link_range, |diag, sp| {
-                                let msg = format!("this item resolved to {} {}, which did not match the disambiguator '{}'", kind.article(), kind.descr(id), disambiguator);
+                                let msg = format!("this link resolved to {} {}, which did not match the disambiguator '{}'", kind.article(), kind.descr(id), disambiguator);
```

Comment:
This is my first time seeing the term "disambiguator". Is there any simpler term?

Reply:
The [nightly docs](https://doc.rust-lang.org/nightly/rustdoc/unstable-features.html#linking-to-items-by-name) say

> In case of ambiguity, rustdoc will warn about the ambiguity and ask you to disambiguate, which can be done by using a prefix like struct@, enum@, type@, trait@, union@, const@, static@, value@, function@, mod@, fn@, module@, method@ , macro@, or derive@

'prefix' seems more ambiguous to me without being less jargony. Do you have any ideas? I like 'disambiguator' because it describes what it does: it removes ambiguity wrt the namespace.

Reply:
> 'prefix' seems more ambiguous to me without being less jargony. Do you have any ideas? I like 'disambiguator' because it describes what it does: it removes ambiguity wrt the namespace.

No idea for a simpler term, I get the meaning after thinking about it for a while but I don't know if that applies to others. Of course, if we have a "did you mean" then we can keep this "disambiguator" since people will likely just apply the fix.

Reply:
I would call it the disambiguator and point it out in the source. It's not a prefix since many of these are suffixes.

Reply:
> Of course, if we have a "did you mean" then we can keep this "disambiguator" since people will likely just apply the fix.

Ooh, this is a really good idea, we could suggest changing `struct@my_enum` to `enum@my_enum` 😄

Reply:
Done!